File: src/ds-visuals/stack/code/stack.c
#include<stdio.h>
#include<stdlib.h>
#include<stdarg.h>
#define MAX 10
//stacks using arrays.

typedef struct stk{
	int arr[MAX];
	int top;
} Stack;

void push(Stack* stack, int element){
	if(stack->top == MAX-1)
		printf("Error- stack full");
	else
		stack->arr[++(stack->top)] = element;
	return;	
}

int pop(Stack*stack){
	if(stack->top == -1)
		printf("Error- Stack empty");
	else
		return stack->arr[(stack->top)--];	
	return -1;
}

void display(Stack *stack){
	int i;
	printf("stack: ");
	for(i = 0; i <= stack->top; i++)
		printf("%d ",stack->arr[i]);
	printf("\n");
	return;
}

Stack createStack(int num_elements, ...){
	Stack stack;
	int i;
	va_list list;
	stack.top = -1;
    va_start(list, num_elements); 
    for (i = 1; i <= num_elements; i++) 
        push(&stack, va_arg(list, int)); 
    va_end(list); 
    return stack;  
}

int main(){
	Stack stack = createStack(3, 100, 200, 300);
	display(&stack);
	printf("element popped: %d\n",pop(&stack));
	display(&stack);
	push(&stack, 400);
	display(&stack);
	return 0;
}

File: src/ds-visuals/stack/code/stack.java
class Stack{
	final int MAX = 10;
	int stack[] = new int[MAX];
	int top = -1;

	Stack(int...arg){
		this.createStack(arg);
	}

	void push(int element){
		if(this.top == MAX-1)
			System.out.println("Error- stack full");
		else
			this.stack[++top] = element;
		return;
	}

	int pop(){
		if(this.top == -1)
			System.out.println("Error- stack empty");
		else
			return this.stack[top--];
		return -1;
	}

	void createStack(int ... arg){
		for(int i = 0; i < arg.length; i++)
			this.push(arg[i]);
		return;
	}

	void display(){
	    System.out.printf("stack: ");
		for(int i = 0; i <= this.top; i++)
			System.out.printf(this.stack[i]+" ");
		System.out.println();
	}
}

class Main{
	public static void main(String args[]){
		Stack s = new Stack(100, 200, 300);
		s.display();
		System.out.println("element popped: "+s.pop());
		s.display();
	    s.push(400);
		s.display();
	}	
}
